### PR TITLE
Blog

### LINKED ISSUE #3: blogs page
complete by 13 th Aug

### PR DESCRIPTION
**Pull request by riyaaaa123**
fix #3

### CODE REVIEW

File: backend/Aries/settings/common.py [hunk 1]
```diff
@@ -18,23 +19,28 @@ class Common(Configuration):
...
 
-    ALLOWED_HOSTS = []
+    ALLOWED_HOSTS = ["*"]
 
 
     # Application definition
 
     INSTALLED_APPS = [
+       
+        'blog.apps.BlogConfig',
```
Comment (pranav-leo) [inline backend/Aries/settings/common.py]: put it in end where all apps are put

File: backend/blog/migrations/0001_initial.py
Comment (pranav-leo) [inline backend/blog/migrations/0001_initial.py:1]: merge all the migrations into one

File: backend/blog/models.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+class Blog(models.Model):
+    title=models.CharField(max_length=200)
+    content= models.TextField()
+    writer= models.ForeignKey(User,on_delete=models.CASCADE,related_name='blog')
+    date_posted= models.DateTimeField(default=timezone.now)
+    image_1= models.ImageField(upload_to=get_blog_pics_path())
+    Web_image= models.ImageField(upload_to=get_blog_pics_path())
+    mobile_image= models.ImageField(upload_to=get_blog_pics_path())
+    isverified= models.IntegerField(default=1,choices=Verification_choices)
+    @classmethod
```
Comment (pranav-leo) [inline backend/blog/models.py]: remove trigram search from this file

File: frontend/package.json [hunk 3]
```diff
@@ -18,22 +18,23 @@
...
-    "autoprefixer": "10.4.14",
+    "@tanstack/react-query": "^4.33.0",
+    "@types/node": "^20.4.8",
+    "@types/react": "^18.2.18",
+    "@types/react-dom": "^18.2.7",
+    "autoprefixer": "^10.4.14",
     "axios": "^1.4.0",
-    "eslint": "8.46.0",
-    "eslint-config-next": "13.4.12",
+    "eslint": "^8.46.0",
```
Comment (pranav-leo) [inline frontend/package.json]: remove ^ added in this page

File: frontend/pages/blog/[id].tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    mutate();
+   },[])
+
+   const blog = blogdata.find((blog) => blog.id ==id);
+   console.log(blog)
+   if (!blog) {
+    return <div>Blog post not found</div>;
+  }
+    
+    const styleObj = {
```
Comment (pranav-iitr) [inline frontend/pages/blog/[id].tsx:33]: border radious jaise cheez ko tailwind use karo and rest ko common CSS file me daldo

File: frontend/pages/blog/index.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  const [posts, setPosts] = useState([]);
+  const [results,setResults]= useState([]);
+  const {mutate,isLoading} = BlogApi((res)=>{ 
+    console.log('Mutation successful!', res);
+      // Update the posts state with the updated data from the response
+      setResults(res.data);
+  },(err)=>{
+    console.log('Mutation error:', err);
+  })
+    // console.log("hihi")
```
Comment (pranav-leo) [inline frontend/pages/blog/index.tsx]: remove unused comments

File: frontend/pages/gallery/index.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+const Gallery = () => {
+    const styleObj = {
+        background: ";",
+        borderRadius: "1.5rem",
+        boxShadow: "2px 2px 2px 0px rgba(255, 255, 255, 0.05) inset, -1px -1px 4px 0px rgba(255, 255, 255, 0.05) inset, 0px 4px 4px 0px rgba(0, 0, 0, 0.10);"
+    }
+    const bg = {
+        background: "rgba(170, 170, 170, 0.30)",
+        backgroundBlendMode: "luminosity",
```
Comment (pranav-iitr) [inline frontend/pages/gallery/index.js:13]: gallery ke liye nai branch banalo

File: frontend/utils/network.js [hunk 7]
```diff
@@ -91,12 +91,13 @@ export const getErrorBody = (error) => {
...
-  if (env == 'production') return process.env.PRODUCTION_SERVER;
-  else if (env == 'staging')
-    return process.env.STAGING_SERVER;
+  // let env = process.env.NODE_ENV;
+  // if (env == 'production') return process.env.PRODUCTION_SERVER;
+  // else if (env == 'staging')
+  //   return process.env.STAGING_SERVER;
   
-  return process.env.DEVELOPMENT_SERVER;
+  return "http://localhost:8000";
```
Comment (pranav-leo) [inline frontend/utils/network.js]: isko env se read karo

File: frontend/package-lock.json
Comment (pranav-iitr) [inline frontend/package-lock.json:1]: remove this file

File: frontend/hooks/blogs.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+const SearchData=(query)=>{
+  const url= getUrl(`/blogs/trigram-search/?q=${encodeURIComponent(query)}`)
+  console.log('Search query in SearchData:', query);
+  return request("GET",url,null,false)
+  
+}
+
+export const BlogApi = (successCallback, errorCallback) => {
+    return useMutation(BlogData, {
+      mutationKey: "get-auth-token",
```
Comment (pranav-iitr) [inline frontend/hooks/blogs.js:18]: mutation key sahi se daalo
and function name ache se likho

File: backend/user/views.py [hunk 9]
```diff
@@ -10,7 +10,7 @@
 from .models import Team_Member
 from  .serilizers import TeamMemberSerializer
 
-class TeamMemberView(ApplicationBaseView,ListCreateAPIView):
+class TeamMemberView(ListCreateAPIView):
```
Comment (pranav-leo) [inline backend/user/views.py]: ApplicationBaseView ko kyu remove kiya ?

File: frontend/hooks/blogs.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
+import {useMutation} from "@tanstack/react-query";
+import { getUrl,request,getErrorBody } from "../utils/network";
+
+const BlogData = () => {
+    const url = getUrl('/blogs/')
+    return request("GET",url,null,false)
+
+}
+const SearchData=(query)=>{
+  const url= getUrl(`/blogs/trigram-search/?q=${encodeURIComponent(query)}`)
```
Comment (pranav-leo) [inline frontend/hooks/blogs.js]: add query params use karo from network

File: backend/utils/constants.py [hunk 11]
```diff
@@ -0,0 +1,6 @@
+Verification_choices=(
+    (0,'Not Verified'),
+    (1,'Verified'),
```
Comment (pranav-leo) [inline backend/utils/constants.py]: use classes

File: backend/blog/models.py [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+# Create your models here.
+class Blog(models.Model):
+    title=models.CharField(max_length=200)
+    content= models.TextField()
+    writer= models.ForeignKey(User,on_delete=models.CASCADE,related_name='blog')
+    date_posted= models.DateTimeField(default=timezone.now)
+    image_1= models.ImageField(upload_to=get_blog_pics_path())
+    Web_image= models.ImageField(upload_to=get_blog_pics_path())
+    mobile_image= models.ImageField(upload_to=get_blog_pics_path())
+    isverified= models.IntegerField(default=1,choices=Verification_choices)
```
Comment (pranav-leo) [inline backend/blog/models.py]: postive integers use kar rahe ho to postive integers use karo

File: backend/blog/models.py [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+from django.utils import timezone
+from utils.constants import Verification_choices
+from utils.path import get_blog_pics_path
+from django.contrib.postgres.search import TrigramSimilarity
+from django.db.models import Q
+from django.contrib.auth import get_user_model
+User = get_user_model()
+
+# Create your models here.
+class Blog(models.Model):
```
Comment (pranav-leo) [inline backend/blog/models.py]: utils se json model and timestamp model use karo

File: backend/Aries/settings/common.py [hunk 14]
```diff
@@ -115,7 +122,11 @@ class Common(Configuration):
     # https://docs.djangoproject.com/en/4.2/howto/static-files/
 
     STATIC_URL = 'static/'
-
+    MEDIA_ROOT = os.path.join(BASE_DIR,'media')
+    MEDIA_URL = 'media/'
+    CORS_ALLOWED_ORIGINS = [
```
Comment (pranav-leo) [inline backend/Aries/settings/common.py]: CORS ki requierment nahi hai

**Review by pranav-leo — CHANGES_REQUESTED** [review]

File: .env.local
Comment (pranav-leo) [inline .env.local:1]: Doesn't needs to be deleted

File: backend/Aries/settings/common.py [hunk 15]
```diff
@@ -10,45 +12,61 @@
 # See https://docs.djangoproject.com/en/4.2/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-from configurations import Configuration
```
Comment (pranav-leo) [inline backend/Aries/settings/common.py:13]: line is requiered

File: backend/Aries/settings/common.py [hunk 16]
```diff
@@ -110,18 +139,27 @@ class Common(Configuration):
...
 
     DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
 
     AUTH_USER_MODEL = 'user.User'
 
-    
+    REST_FRAMEWORK = {
+        'DEFAULT_SCHEMA_CLASS': 'rest_framework.schemas.coreapi.AutoSchema'}
 
+    # CACHE_MIDDLEWARE_ALIAS = 'default'  # The cache alias to use for storage and 'default' is **local-memory cache**.
```
Comment (pranav-leo) [inline backend/Aries/settings/common.py]: remove these lines

File: backend/blog/urls.py [hunk 17]
```diff
@@ -0,0 +1,8 @@
+from django.urls import path
+from .views import BlogView,BlogCount
+
+urlpatterns = [
+    path('', BlogView.as_view(),name='blog-list'),
+    path('blogs/<int:blog_id>', BlogCount.as_view(), name='blog-count'),
+    # path('trigram-search/',trigram_search_view.as_view(), name='trigram_search'),
```
Comment (pranav-leo) [inline backend/blog/urls.py]: remove un used lines

File: backend/utils/constants.py [hunk 18]
```diff
@@ -0,0 +1,13 @@
+class VerificationStatus:
+    PENDING = 1
+    APPROVED = 2
+    REJECTED = 3
+    IN_PROGRESS = 4
+
+VERIFICATION_STATUS_CHOICES = (
```
Comment (pranav-leo) [inline backend/utils/constants.py]: will be a member of class

File: frontend/pages/blog/[id].tsx [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+  
+  };
+
+  const { id } = router.query;
+  console.log("ID from URL:", id); 
+  const [data, setData] = useState<BlogData|null>(null);
+
+  const {mutate,isLoading} = SingleBlogApi(id,
+     (res)=> { 
+    console.log('Mutation successful!', res.data);
```
Comment (pranav-leo) [inline frontend/pages/blog/[id].tsx]: review console logs

File: frontend/pages/blog/index.tsx [hunk 20]
```diff
@@ -0,0 +1,120 @@
+import React, {useState,useEffect} from "react";
+import {BlogApi} from "../../hooks/blogs";
+import styles from "../../styles/Blog.module.css";
+import Link from 'next/link';
+
+const Blogs=()=>{
+  const pageSize = 10;
+  const [searchText, setSearchText] = useState('');
```
Comment (pranav-leo) [inline frontend/pages/blog/index.tsx]: type script ke formats sahi se use karo

File: frontend/pages/gallery/index.js [hunk 21]
```diff
@@ -0,0 +1,65 @@
+import React , { useState } from 'react'
```
Comment (pranav-leo) [inline frontend/pages/gallery/index.js]: remove code for gallery

File: frontend/utils/network.js [hunk 22]
```diff
@@ -91,12 +92,12 @@ export const getErrorBody = (error) => {
 };
 
 const get_base_api = () => {
-  let env = process.env.NODE_ENV;
-  if (env == 'production') return process.env.PRODUCTION_SERVER;
-  else if (env == 'staging')
-    return process.env.STAGING_SERVER;
+  // let env = process.env.NODE_ENV;
+  // if (env == 'production') return process.env.PRODUCTION_SERVER;
```
Comment (pranav-leo) [inline frontend/utils/network.js:92]: read these form env

**Review by pranav-leo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**